### PR TITLE
[fix](Nereids) fix lexer Backtracking or Ambiguity cause of key word duplicate

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
nereids timeout cause of parser cost too long time
intro by： https://github.com/apache/doris/pull/39113

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 [hunk 1]
```diff
@@ -703,11 +688,11 @@ SIMPLE_COMMENT
     ;
 
 BRACKETED_COMMENT
-    : '/*' {!isHint()}? ( BRACKETED_COMMENT | . )*? ('*/' | {markUnclosedComment();} EOF) -> channel(HIDDEN)
+    : COMMENT_START ( BRACKETED_COMMENT | . )*? ('*/' | {markUnclosedComment();} EOF) -> channel(HIDDEN)
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4:691]:
```suggestion
    : COMMENT_START ( BRACKETED_COMMENT | HINT_WITH_CHANNEL | . )*? ('*/' | {markUnclosedComment();} EOF) -> channel(HIDDEN)
```

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 [hunk 2]
```diff
@@ -624,6 +608,7 @@ COLON: ':';
 ARROW: '->';
 HINT_START: '/*+';
 HINT_END: '*/';
+COMMENT_START: '/*';
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4:611]: add to non-reserved key word

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 [hunk 3]
```diff
@@ -703,11 +688,11 @@ SIMPLE_COMMENT
     ;
 
 BRACKETED_COMMENT
-    : '/*' {!isHint()}? ( BRACKETED_COMMENT | . )*? ('*/' | {markUnclosedComment();} EOF) -> channel(HIDDEN)
+    : COMMENT_START ( BRACKETED_COMMENT | . )*? ('*/' | {markUnclosedComment();} EOF) -> channel(HIDDEN)
     ;
 
 HINT_WITH_CHANNEL
-    : {isChannel2()}? HINT_START .*? HINT_END -> channel(2)
+    : HINT_START {isChannel2()}? .*? HINT_END -> channel(2)
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4:695]:
```suggestion
    : HINT_START {isChannel2()}? ( BRACKETED_COMMENT | HINT_WITH_CHANNEL | . )*? HINT_END -> channel(2)
```

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 [hunk 4]
```diff
@@ -703,11 +688,11 @@ SIMPLE_COMMENT
     ;
 
 BRACKETED_COMMENT
-    : '/*' {!isHint()}? ( BRACKETED_COMMENT | . )*? ('*/' | {markUnclosedComment();} EOF) -> channel(HIDDEN)
+    : COMMENT_START ( BRACKETED_COMMENT | . )*? ('*/' | {markUnclosedComment();} EOF) -> channel(HIDDEN)
     ;
 
 HINT_WITH_CHANNEL
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4:694]:
```suggestion
HINT
```

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 (hunk 1 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4]: done

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 (hunk 3 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4]: done

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 (hunk 4 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4:694]: done

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 (hunk 2 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4:611]: done

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 5]
```diff
@@ -1267,6 +1267,7 @@ nonReserved
     | COLOCATE
     | COLUMNS
     | COMMENT
+    | COMMENT_START
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:1270]: add HINT_START HINT_END too

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run feut

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 [hunk 6]
```diff
@@ -702,12 +687,12 @@ SIMPLE_COMMENT
...
-BRACKETED_COMMENT
-    : '/*' {!isHint()}? ( BRACKETED_COMMENT | . )*? ('*/' | {markUnclosedComment();} EOF) -> channel(HIDDEN)
+HINT
+    : {isChannel()}? HINT_START ( BRACKETED_COMMENT | HINT | . )*? HINT_END -> channel(2)
     ;
 
-HINT_WITH_CHANNEL
-    : {isChannel2()}? HINT_START .*? HINT_END -> channel(2)
+BRACKETED_COMMENT
+    : {isChannel()}? COMMENT_START ( BRACKETED_COMMENT | HINT | . )*? ('*/' | {markUnclosedComment();} EOF) -> channel(HIDDEN)
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4:695]: why add isChannel here?

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run feut

**Comment by LiBinfeng-01** [comment]
run feut

**Comment by LiBinfeng-01** [comment]
run cloud_p1

**Comment by LiBinfeng-01** [comment]
run cloud_p0

**Comment by LiBinfeng-01** [comment]
run feut

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**